web-admin/src/main/java/pl/edu/icm/unity/webadmin/tprofile/RuleComponent.java

```diff
@@ -341,11 +357,41 @@ private void setReadOnlyStyle(boolean readOnly)
 		condition.setReadOnly(readOnly);
 		actionEditor.setReadOnlyStyle(readOnly);
 	}
+	
+	private void showHideContent(boolean show)
+	{
+		showHide.setIcon(show ? Images.vaadinUpArrow.getResource()
+				: Images.vaadinDownArrow.getResource());
+		content.setVisible(show);
+	}
+
+	public static class DragHtmlLabel extends Label
+	{
+		RuleComponent parentRule;
+		public DragHtmlLabel(RuleComponent parent, String value)
+		{
+			super(value, ContentMode.HTML);
+			this.parentRule = parent;
+		}
+		
+		public RuleComponent getParentRule()
+		{
+			return parentRule;
+		}
+	}	
+	
+	private final class CallbackImplementation implements pl.edu.icm.unity.webadmin.tprofile.ActionEditor.Callback
+	{
 
+		@Override
+		public void refresh(ActionEditor editor)
+		{
+			info.setValue(editor.getStringRepresentation());
```

Comment:
It would be nicer if the event contain just the string. Passing Action editor creates tight coupling with large objects - in general UI component should never be an event.